docker/Dockerfile

```diff
@@ -1,27 +1,35 @@
-FROM alpine:3.10
+FROM alpine:3.11 AS downloader
 
-ENV APP_VERSION=5.9.0
-ENV PATH="/mattermost-push-proxy/bin:${PATH}"
-
-# Install some needed packages
-RUN apk add --no-cache\
-      netcat-openbsd \
-      ca-certificates
+ENV MM_PROXY_PACKAGE=https://github.com/mattermost/mattermost-push-proxy/releases/download/v5.21/mattermost-push-proxy.tar.gz
 
 RUN mkdir -p /mattermost-push-proxy \
-    && cd /mattermost-push-proxy \
-    && wget https://github.com/mattermost/mattermost-push-proxy/releases/download/v$APP_VERSION/mattermost-push-proxy-$APP_VERSION.tar.gz \
-    && tar -xvzf mattermost-push-proxy-$APP_VERSION.tar.gz \
-    && rm mattermost-push-proxy-$APP_VERSION.tar.gz \
-    && mv mattermost-push-proxy/* ./ \
-    && rm -rf mattermost-push-proxy
+    && wget $MM_PROXY_PACKAGE \
+    && tar -C /mattermost-push-proxy -xzvf mattermost-push-proxy.tar.gz
+
+FROM alpine:3.11 AS runner
+
+RUN apk add --no-cache \
+    ca-certificates \
+    libc6-compat \
+    libffi-dev \
+    linux-headers \
+    netcat-openbsd \
+    tzdata \
+    && rm -rf /tmp/*
+
+COPY --from=downloader /mattermost-push-proxy/* /mattermost-push-proxy/
 
 WORKDIR /mattermost-push-proxy
 
-COPY entrypoint.sh /
+ENV PUSH_PROXY=/mattermost-push-proxy/bin/mattermost-push-proxy \
+    USER_UID=10001 \
```

Comment:
Any reason why we are explicitly specifying the user id ? Shouldn't the `adduser` command be sufficient for this ?

This should make the entire user_setup stage unnecessary.

Reply:
we can do that as well, just add that because we have for the other images and i got that from a best practice from redhat